torchgeo/datasets/digital_typhoon.py

```diff
@@ -0,0 +1,454 @@
+    for more information about the dataset.
+
+    Dataset features:
+
+    * infrared channel images from the Himawari weather satellite (512x512 px)
+      at 5km spatial resolution
+    * auxiliary features such as wind speed, pressure, and more that can be used
+      for regression or classification tasks
+    * 1,099 typhoons and 189,364 images
+
+    Dataset format:
+
+    * hdf5 files containing the infrared channel images
+    * .csv files containing the metadata for each image
+
+    If you use this dataset in your research, please cite the following papers:
+
+    * https://doi.org/10.20783/DIAS.664
+
+    .. versionadded:: 0.6
+    """
+
+    valid_tasks: ClassVar[list[str]] = ['classification', 'regression']
+    aux_file_name: ClassVar[str] = 'aux_data.csv'
+
+    valid_features: ClassVar[list[str]] = [
+        'year',
+        'month',
+        'day',
+        'hour',
+        'grade',
+        'lat',
+        'lng',
+        'pressure',
+        'wind',
+        'dir50',
+        'long50',
+        'short50',
+        'dir30',
+        'long30',
+        'short30',
+        'landfall',
+        'intp',
+    ]
+
+    url: ClassVar[str] = (
+        'https://huggingface.co/datasets/torchgeo/digital_typhoon/resolve/main/WP.tar.gz{0}'
+    )
+
+    md5sums: ClassVar[dict[str, str]] = {
+        'aa': '3af98052aed17e0ddb1e94caca2582e2',
+        'ab': '2c5d25455ac8aef1de33fe6456ab2c8d',
+    }
+
+    min_input_clamp = 170.0
+    max_input_clamp = 300.0
+
+    data_root = 'WP'
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        task: str = 'regression',
+        features: Sequence[str] = ['wind'],
+        targets: list[str] = ['wind'],
+        sequence_length: int = 3,
+        min_feature_value: dict[str, float] | None = None,
+        max_feature_value: dict[str, float] | None = None,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Digital Typhoon Analysis dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            task: whether to load 'regression' or 'classification' labels
+            features: which auxiliary features to return
+            targets: which auxiliary features to use as targets
+            sequence_length: length of the sequence to return
+            min_feature_value: minimum value for each feature
+            max_feature_value: maximum value for each feature
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``task`` argument is invalid
```

Comment:
Or a couple other arguments:
```suggestion
            AssertionError: If any arguments are invalid.
```